tests/Fixtures/ObjectWithEnums.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Tests\Fixtures;
+
+use JMS\Serializer\Annotation as Serializer;
+use JMS\Serializer\Tests\Fixtures\Enum\BackedSuit;
+use JMS\Serializer\Tests\Fixtures\Enum\BackedSuitInt;
+use JMS\Serializer\Tests\Fixtures\Enum\Suit;
+
+class ObjectWithEnums
+{
+    /**
+     * @Serializer\Type("enum<'JMS\Serializer\Tests\Fixtures\Enum\Suit', 'name'>")
+     */
+    public Suit $ordinary;
+    /**
+     * @Serializer\Type("enum<'JMS\Serializer\Tests\Fixtures\Enum\BackedSuit', 'value'>")
+     */
+    public BackedSuit $backed;
+
+    public Suit $ordinaryAutoDetect;
+
+    public BackedSuit $backedAutoDetect;
+
+    public BackedSuitInt $backedIntAutoDetect;
+
+    /**
+     * @Serializer\Type("enum<'JMS\Serializer\Tests\Fixtures\Enum\BackedSuitInt', 'value'>")
+     */
+    public BackedSuitInt $backedInt;
+
+    /**
+     * @Serializer\Type("enum<'JMS\Serializer\Tests\Fixtures\Enum\BackedSuit', 'name'>")
+     */
```

Comment:
Would be great to also have a testcase that uses Attributes.

Reply:
unfortunately the `@Serializer\Type` annotation takes only one param as a string, so except of doing string interpolations, there is no other way to make this wortk.

but that is something that would be interesting to refactor, adding support for something like

`#Type(name: BackedSuit::class, params: [....])`

Reply:
@goetas Hi. I'm trying to use your branch to check how it works.
Could you give an example of how to use `VirtualProperty` along with enum `Type`?

Reply:
Or create a specific EnumType attribute? 

```
#[EnumType(BackedSuit::class)
```

It can then also have more properties for the `value` / `integer` thing. 

No need to support Annotations for this, as most people that use an PHP 8.0 enum are also people that already upgraded to Attributes 😉

Reply:
Any solution with attributes for this case?
```php
#[Type(name: 'list<enum<MyEnum>>')]
```

Reply:
Or composable attributes like this:
`#[Type(new ListType(new EnumType(MyEnum::class))]`

Reply:
changing the `@Type` syntax of out of the scope of the current PR (of course i will accept a PR that improves that).

I would like to see something as 

```php
#[Type('enum', [MyEnum::class]])]
#[Type('array, [['enum', [MyEnum::class]]])]
#[Type('array, ['string', ['enum', [MyEnum::class]]])]
```

that is equivalent to

```
@Type("enum<MyEnum>")
@Type("array<enum<MyEnum>>")
@Type("array<enum<string,MyEnum>>")
```

maybe there is a mess with `[]`....

Reply:
I think I'd suggest a syntax like this. It's consistent in how generic types are noted no matter how deep you go.

```php
#[Type('enum', new Type(MyEnum::class))]
#[Type('array', new Type('enum', new Type(MyEnum::class)))]
#[Type('array', 'string', new Type('enum', new Type(MyEnum::class)))]
```

Reply:
I agree with @enumag. When designing the Attribute syntax, take full advantage of the possibilities there.

Reply:
@ruudk your solution could be as aproxy on top of @enumag's solution

Reply:
>I think I'd suggest a syntax like this using nested attributes, variadic parameter and union type.

that looks very good